ddsrouter_utils/include/ddsrouter_utils/types/impl/Fuzzy.ipp

```diff
@@ -0,0 +1,164 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file Fuzzy.ipp
+ *
+ */
+
+#ifndef _DDSROUTERUTILS_TYPES_FUZZY_IMPL_IPP_
+#define _DDSROUTERUTILS_TYPES_FUZZY_IMPL_IPP_
+
+namespace eprosima {
+namespace ddsrouter {
+namespace utils {
+
+template <typename T>
+Fuzzy<T>::Fuzzy(
+        const T& other,
+        FuzzyLevelType level /* = FuzzyLevelValues::set */)
+    : value(other)
+    , fuzzy_level(level)
+{
+}
+
+template <typename T>
+Fuzzy<T>::Fuzzy(
+        T&& other,
+        FuzzyLevelType level /* = FuzzyLevelValues::set */)
+    : value(std::move(other))
+    , fuzzy_level(level)
+{
+}
+
+/////////////////////////
+// OPERATORS
+/////////////////////////
+
+template <typename T>
+Fuzzy<T>::operator const T&() const noexcept
+{
+    return value;
+}
+
+template <typename T>
+bool Fuzzy<T>::operator==(const Fuzzy<T>& other) const noexcept
+{
+    // If both unset, the object is the same
+    if (!this->is_set() && !other.is_set())
+    {
+        return true;
+    }
+    else
+    {
+        return this->fuzzy_level == other.fuzzy_level || this->value == other.value;
```

Comment:
This depends on what the doxygen says :sweat_smile: 
The point here is what we want to compare, the object `Fuzzy` or the internal object? There is an `operator==(T)` already.
I am not sure what we would prefer, but probably it would be better to only check the value.